rllib/agents/dqn/dqn.py

```diff
@@ -226,6 +176,9 @@ def execution_plan(workers: WorkerSet,
         replay_burn_in=config.get("burn_in", 0),
         replay_zero_init_states=config.get("zero_init_states", True),
         **prio_args)
+    # Assign to Trainer, so we can store the LocalReplayBuffer's
+    # data when we save checkpoints.
+    trainer._local_replay_buffer = local_replay_buffer
```

Comment:
Hmm, yeah, we should probably put the buffer construction into `before_init` (`trainer_template.py::build_trainer()`). We can then check inside the execution plan, whether a buffer has been properly constructed and exists under some trainer property. It's not that much different, though, from what we do here, unless you say that we have a base `Trainer.replay_buffer` property (similar to how we have a `trainer.workers`, `policy.model`, etc..).

Reply:
oh totally not a comment about functionality.
I am just trying to suggest we avoid setting "private" member variables outside of the class itself. since that would make it harder to answer the question like: where does this local_replay_buffer come from.
it's totally fine for now. we plan on refactoring a bunch of things anyways.